google/docs/update_doc.py

```diff
@@ -0,0 +1,145 @@
+                    },
+                    "textStyle": text_style,
+                    "fields": ",".join(text_style.keys())
+                }
+            })
+
+        # Update the current index to account for the added text
+        current_index += text_length
+
+    # Process elements in the markdown
+    for element in soup.contents:
+        if element.name in ['p']:
+            add_text_request(element.get_text() + "\n")
+        elif element.name in ['h1', 'h2', 'h3']:
+            # Apply bold style for headers
+            add_text_request(element.get_text() + "\n", bold=True)
+        elif element.name in ['ul']:
+            for li in element.find_all('li'):
+                add_text_request("\u2022 " + li.get_text() + "\n")
+        elif element.name in ['ol']:
+            for i, li in enumerate(element.find_all('li'), start=1):
+                add_text_request(f"{i}. " + li.get_text() + "\n")
+        elif element.name == 'a':
+            # Add link
+            add_text_request(element.get_text(), link=element['href'])
+        elif element.name == 'table':
+            for row in element.find_all('tr'):
+                row_text = "\t".join([cell.get_text() for cell in row.find_all(['td', 'th'])]) + "\n"
+                add_text_request(row_text)
+        else:
+            # Default handling for unknown elements
+            add_text_request(element.get_text() + "\n")
+
+    return requests
+
+
+def main():
+    try:
+        doc_ref = os.getenv('DOC_REF')
+        new_doc_content = os.getenv('NEW_DOC_CONTENT')
+
+        if not doc_ref:
+            raise ValueError('DOC_REF environment variable is missing or empty')
+
+        if not new_doc_content:
+            raise ValueError('NEW_DOC_CONTENT environment variable is missing or empty')
+
+        try:
+            requests = markdown_to_google_doc_requests(new_doc_content)
+        except Exception as e:
+            raise ValueError(f"Failed to parse NEW_DOC_CONTENT: {e}")
+
+        file_id = extract_file_id(doc_ref)
+        service = client('docs', 'v1')
+
+        # Retrieve the document to determine its length
+        document = service.documents().get(documentId=file_id).execute()
+        content = document.get('body').get('content')
+        document_length = content[-1].get('endIndex') if content and 'endIndex' in content[-1] else 1
+
+        if document_length > 2:
+            # Clear existing document content
+            clear_requests = [
+                {
+                    "deleteContentRange": {
+                        "range": {
+                            "startIndex": 1,
+                            "endIndex": document_length - 1
+                        }
+                    }
+                }
+            ]
+
+            # Issue a batch update request to clear the document
+            service.documents().batchUpdate(
+                documentId=file_id,
+                body={"requests": clear_requests}
+            ).execute()
```

Comment:
Can we make both of these batch requests at once?

Reply:
yeah, that's probably better. need to test if that causes issues with the doc indexing first.